DeepDiveLogCompiler.scala

```diff
@@ -235,147 +281,168 @@ object DeepDiveLogCompiler extends DeepDiveLogHandler {
   type CompiledBlocks = List[CompiledBlock]
 
   // Dispatch to the corresponding compile function
-  def compile(stmt: Statement, ss: CompilationState): CompiledBlocks = stmt match {
-    case s:  ExtractionRule   => compile(s, ss)
-    case s:  FunctionCallRule => compile(s, ss)
-    case s:  InferenceRule    => compile(s, ss)
-    case _                    => List()  // defaults to compiling into empty block
-  }
+  // def compile(stmts: List[Statement], ss: CompilationState, isIncremental: Boolean): CompiledBlocks = stmts(0) match {
+  //   case s: ExtractionRule   => compileE(stmts, ss, isIncremental)
+  //   case s: FunctionCallRule => compileF(stmts, ss, isIncremental)
+  //   case s: InferenceRule    => compileI(stmts, ss, isIncremental)
+  //   case _                   => List()  // defaults to compiling into empty block
+  // }
 
   // Generate extraction rule part for deepdive
-  def compile(stmt: ExtractionRule, ss: CompilationState): CompiledBlocks = {
-    // Generate the body of the query.
-    val qs              = new QuerySchema( stmt.q )
-    // variable columns
-    val variableCols = stmt.q.head.terms flatMap {
-      case(Variable(v,rr,i)) => ss.resolveColumn(v, qs, stmt.q, true)
-    }
+  def compileE(stmts: List[ExtractionRule], ss: CompilationState, isIncremental: Boolean): CompiledBlocks = {
```

Comment:
Let's keep the compile function signature simple: isIncremental should be passed indirectly as a member to CompilationState, because that's exactly why we created the CompilationState class and folded everything into it.

Also, let's keep the function names descriptive, e.g., `compileExtractionRules` instead of `compileE`.  It's called only once below anyway, so we're not paying any price for a clearer name.